### PR TITLE
Add dash fixed header

### PR DESCRIPTION
**Pull request by flops**
Flag: EnableDashFixedHeader

### CODE REVIEW

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9881-dash-fixed-header-1240/dash-main-view-1240/ru/">dash.main.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9881-dash-fixed-header-1240/dash-main-view-1240/en/">dash.main.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9881-dash-fixed-header-1240/dash-main-view-1240/ru/">dash.main.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9881-dash-fixed-header-1240/dash-main-view-1240/en/">dash.main.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>Errors occurred when merging changes from your branch with the Weblate branch.</summary>

```
Rebasing (1/1)
Auto-merging src/ui/units/dash/containers/Body/Body.tsx
CONFLICT (content): Merge conflict in src/ui/units/dash/containers/Body/Body.tsx
error: could not apply 100c8892... Translations changed
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 100c8892... Translations changed
 (1)
```

</details>

**Resolve conflicts according to instructions**
1. Switch to the current branch associated with this pull request.
```
git checkout CHARTS-9881-dash-fixed-header
```
2. Add Weblate as remote. And remove previous upstream with the same name:
```
git remote remove weblate 2> /dev/null
git remote add weblate https://weblate.gravity-ui.com/git/datalens/charts-9881-dash-fixed-header-1240/chartkit-1240/
git -c http.sslVerify=false remote update weblate
```
3. Merge Weblate changes:
```
git merge weblate/CHARTS-9881-dash-fixed-header
```
4. Resolve conflicts:
```
edit ...
git add ...
git commit
```
5. Push changes to upstream repository, Weblate will fetch merge from there:
```
git push origin
```

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9881-dash-fixed-header-1240/dash-main-view-1240/ru/">dash.main.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9881-dash-fixed-header-1240/dash-main-view-1240/en/">dash.main.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

File: src/ui/styles/base.scss [hunk 1]
```diff
@@ -32,4 +32,9 @@ body.js-screenshot-mode * {
 body.g-root {
     --g-font-family-sans: 'YS Text', 'Helvetica Neue', 'Arial', 'Helvetica', sans-serif;
     --gn-aside-top-panel-height: 0px;
+    overscroll-behavior: none;
+}
+
+body.g-root_mobile {
+    overscroll-behavior: auto;
 }
```
Comment (Marginy605) [inline src/ui/styles/base.scss:40]: What for and do we need it for all pages, not only dash?

File: src/ui/styles/variables.scss [hunk 2]
```diff
@@ -55,6 +55,7 @@ $monospaceFontFamily: 'Consolas', 'Menlo', 'Ubuntu Mono', monospace;
         --w-disabled-item-background: rgba(0, 0, 0, 0.07);
         --dl-color-contrast-4: rgba(0, 0, 0, 0.04);
         --dl-color-base-workbook-icon: #d1dfff;
+        --dl-color-fixed-header-background: rgb(242, 242, 242);
```
Comment (Marginy605) [inline src/ui/styles/variables.scss:58]: Will we be able to override by tenant theme this background?

File: src/ui/units/dash/containers/Body/Body.tsx [hunk 3]
```diff
@@ -154,11 +188,50 @@ class Body extends React.PureComponent<BodyProps> {
         });
     }, UPDATE_STATE_DEBOUNCE_TIME);
 
+    _memoizedContext: MemoContext = {};
+    _memoizedControls: DashKitProps['overlayControls'];
+    _memoizedMenu: DashKitProps['overlayMenuItems'];
+    _memoizedWidgetsMap: any = {layout: null, byGroup: {}, byId: {}}; // TODO type
```
Comment (Marginy605) [inline src/ui/units/dash/containers/Body/Body.tsx:194]: TO DO type

File: src/ui/units/dash/containers/FixedHeader/FixedHeader.scss [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    &__container-wrapper {
+        display: flex;
+        padding-bottom: 8px;
+        flex-direction: row;
+        flex: 1;
+
+        > .react-grid-layout {
+            flex: 1;
+            transition: none;
+        }
```
Comment (Marginy605) [inline src/ui/units/dash/containers/FixedHeader/FixedHeader.scss:72]: It's better add one mixin for part
```
> .react-grid-layout {
            flex: 1;
            transition: none;
        }
```
and another for 
```
> .react-grid-layout {
            flex: 1;
            transition: none;
        }
```
Both mixins could be local or in common components style file

**Review by Marginy605 — CHANGES_REQUESTED** [review]

**Review by Marginy605 — APPROVED** [review]

**Pull request merged**